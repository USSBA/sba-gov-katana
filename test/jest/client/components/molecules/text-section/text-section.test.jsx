/*global expect*/

import React from 'react';
import TextSection from 'client/components/molecules/text-section/text-section.jsx';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

test('TextSection renders a simple paragraph', () => {
  let text = "<p>test</p>";
  const component = renderer.create(<TextSection text={text} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('TextSection renders several paragraph', () => {
  let text = "<p>test</p><p>test</p><p>test</p>";
  const component = renderer.create(<TextSection text={text} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('TextSection renders several paragraphs with an unordered list in the middle', () => {
  let text = "<p>test</p><ul><li>1</li>1<li></li></ul><p>test</p>";
  const component = renderer.create(<TextSection text={text} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
