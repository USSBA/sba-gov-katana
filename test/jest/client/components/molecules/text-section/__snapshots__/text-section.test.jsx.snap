// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSection renders a simple paragraph 1`] = `
<p
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>test</p>",
    }
  }
/>
`;

exports[`TextSection renders several paragraph 1`] = `
<p
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>test</p><p>test</p><p>test</p>",
    }
  }
/>
`;

exports[`TextSection renders several paragraphs with an unordered list in the middle 1`] = `
<p
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>test</p><ul><li>1</li>1<li></li></ul><p>test</p>",
    }
  }
/>
`;
